### ISSUE TITLE
Automatically register handlers like in .NET

### ISSUE THREAD

**Issue by smjxpro** [GeorgopoulosGiannis/dart_mediatr#11]
I know I know Dart doesn't support reflection or mirror properly. But it's so much cumbersome of registering all the request handlers manually.

Dart is planning to introduce static meta-programming soon. Therefore, this issue will act as a way around of not to having register handlers manually when it's available.

**Comment by georgopoulosgianniss** [comment]
Yes I agree, I am tracking the issue waiting for static meta-programming to arrive. If it takes too long and I have the time maybe I can implement something with the build_runner package

**Comment by smjxpro** [comment]
Hearing rumors that Dart 4 will have it. I hate build_runner BTW. It's nowhere near like .NET code-generation tools like dotnet-ef.

**Comment by GeorgopoulosGiannis** [comment]
I have the same feelings for build_runner,  let's wait for dart 4 and see.
